### PR TITLE
Add more explicit UI for unchanged data for dvc update

### PR DESCRIPTION
**Pull request by karajan1001**
partly fix: #8464
1. Add some UI message in `dvc update` for unchanged data pushed to some worktree remotes

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/experiments/queue/celery.py [hunk 1]
```diff
@@ -348,7 +348,10 @@ def _mark_inactive_tasks_failure(
                         task_id,
                         remained_entries[entry],
                     )
-                    self.celery.backend.mark_as_failure(task_id, None)
+                    backend = self.celery.backend
+                    backend.mark_as_failure(  # type: ignore[attr-defined]
+                        task_id, None
+                    )
```
Comment (dberenbaum) [inline dvc/repo/experiments/queue/celery.py:354]: Is this intended to be in this PR?

Comment (dberenbaum) [reply dvc/repo/experiments/queue/celery.py:354]: Never mind -- I see it's a mypy issue.

**Comment by dberenbaum** [comment]
@karajan1001 Considering the scope has been reduced to exclude worktree remotes, should we merge this one? If so, could you add an example or video of the UI?

**Comment by karajan1001** [comment]
> Considering the scope has been reduced to exclude worktree remotes,

Excuse me any details? 

This is the demo, compares to the third video in https://github.com/iterative/dvc/issues/8464#issuecomment-1387141569
[asciicast](https://asciinema.org/a/EKiZMHfypyJ7iJpSZec3hzaeU)

File: dvc/repo/experiments/queue/remove.py [hunk 2]
```diff
@@ -58,7 +58,8 @@ def remove_tasks(  # noqa: C901
             result: AsyncResult = AsyncResult(task_id)
             if result is not None:
                 result.forget()
-            celery_queue.celery.purge(msg.delivery_tag)
+            if msg.delivery_tag:
+                celery_queue.celery.purge(msg.delivery_tag)
```
Comment (daavoo) [inline dvc/repo/experiments/queue/remove.py:62]: @karajan1001 This changes look unrelated

File: dvc/repo/worktree.py [hunk 3]
```diff
@@ -332,24 +340,32 @@ def _update_worktree_out(
...
         )
-        return
+        return 0
 
-    _fetch_out_changes(out, local_index, remote_index, remote)
-    _update_out_meta(
-        repo,
+    transferred = _fetch_out_changes(out, local_index, remote_index, remote)
+    logger.debug(
+        "Update %s files to out '%s' from worktree of '%s'",
```
Comment (daavoo) [inline dvc/repo/worktree.py:347]:
```suggestion
        "Updated %s files in '%s' from worktree '%s'",
```

**Review by daavoo — APPROVED** [review]
This improves the UI but doesn't seem to fix the actual problem in #8464 (unless we have cut the scope to only the UI)

File: dvc/repo/experiments/queue/remove.py (hunk 2 above)
Comment (karajan1001) [reply dvc/repo/experiments/queue/remove.py:62]: Yes, there might be something wrong with My environment, without these changes I can't pass `pre-commit` check, I will create a separate PR for them

**Comment by dberenbaum** [comment]
@karajan1001 Maybe you missed or misunderstood https://iterativeai.slack.com/archives/CB41NAL8H/p1675437378808469? We are no longer working on `worktree` remotes. I will close this one, but feel free to reopen if there's still useful parts of it.
